metadata-ingestion/src/datahub/ingestion/source/looker/looker_template_language.py

```diff
@@ -92,52 +107,195 @@ def resolve_liquid_variable(text: str, liquid_variable: Dict[Any, Any]) -> str:
+    """
 
-            view["derived_table"]["datahub_transformed_sql"] = _complete_incomplete_sql(
-                raw_view=view, sql=view["derived_table"]["datahub_transformed_sql"]
+    evaluate_to_true_regx: str
+    remove_if_comment_line_regx: str
+
+    def __init__(self, source_config: LookMLSourceConfig):
+        super().__init__(source_config=source_config)
+
+        # This regx will keep whatever after -- if looker_environment --
+        self.evaluate_to_true_regx = r"-- if {} --".format(
+            self.source_config.looker_environment
+        )
+
+        # It will remove all other lines starts with -- if ... --
+        self.remove_if_comment_line_regx = r"-- if {} --.*?(?=\n|-- if|$)".format(
+            dev if self.source_config.looker_environment.lower() == prod else prod
+        )
+
+    def _apply_regx(self, value: str) -> str:
+        result: str = re.sub(
+            self.remove_if_comment_line_regx, "", value, flags=re.IGNORECASE | re.DOTALL
+        )
+
+        # Remove '-- if prod --' but keep the rest of the line
+        result = re.sub(self.evaluate_to_true_regx, "", result, flags=re.IGNORECASE)
+
+        return result
+
+    def _apply_transformation(self, value: str, view: dict) -> str:
+        return self._apply_regx(value)
+
+
+class TransformedLookMlView:
+    transformers: List[LookMLViewTransformer]
+    view_dict: dict
+    transformed_dict: dict
+
+    def __init__(
+        self,
+        transformers: List[LookMLViewTransformer],
+        view_dict: dict,
+    ):
+        self.transformers = transformers
+        self.view_dict = view_dict
+        self.transformed_dict = {}
+
+    def view(self) -> dict:
+        if self.transformed_dict:
+            return self.transformed_dict
+
+        self.transformed_dict = {**self.view_dict}
+
+        logger.debug(f"Processing view {self.view_dict[NAME]}")
+
+        for transformer in self.transformers:
+            logger.debug(f"Applying transformer {transformer.__class__.__name__}")
+
+            self.transformed_dict = always_merger.merge(
+                self.transformed_dict, transformer.transform(self.transformed_dict)
             )
+
+        return self.transformed_dict
+
+
+def process_lookml_template_language(
+    source_config: LookMLSourceConfig,
+    view_lkml_file_dict: dict,
+) -> None:
+    if "views" not in view_lkml_file_dict:
+        return
+
+    transformers: List[LookMLViewTransformer] = [
+        LookMlIfCommentTransformer(
+            source_config=source_config
+        ),  # First evaluate the -- if -- comments. Looker does the same
+        LiquidVariableTransformer(
+            source_config=source_config
+        ),  # Now resolve liquid variables
+        DropDerivedViewPatternTransformer(
+            source_config=source_config
+        ),  # Remove any ${} symbol
+        IncompleteSqlTransformer(
+            source_config=source_config
+        ),  # complete any incomplete sql
+    ]
```

Comment:
interesting - so we're pushing some of our transformation logic into the jinja / lkml parse layer?